### PR TITLE
Fix #2063 - App crashes for blank search query.

### LINKED ISSUE #2063: App Crashes for blank search query
**Actual Behaviour**

App crashes if you type something and then clear the search box.

**Expected Behaviour**

It should not crash and display normal search results for blank query.

**Steps to reproduce it**

Open Search.
Type something.
Clear the query.
Try to scroll through the results.

**LogCat for the issue**

12-26 15:55:05.144 32400-32400/org.fossasia.openevent E/AndroidRuntime: FATAL EXCEPTION: main
                                                                        Process: org.fossasia.openevent, PID: 32400
                                                                        java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid item position 4(offset:4).state:322 android.support.v7.widget.RecyclerView{9697df9 VFED..... ........ 0,0-1080,834 #7f090119 app:id/search_recyclerView}, adapter:org.fossasia.openevent.adapters.GlobalSearchAdapter@8a583d1, layout:android.support.v7.widget.LinearLayoutManager@148fe36, context:org.fossasia.openevent.activities.SearchActivity@e4a3adb
                                                                            at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5654)
                                                                            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5589)
                                                                            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5585)
                                                                            at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2231)
                                                                            at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1558)
                                                                            at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1518)
                                                                            at android.support.v7.widget.LinearLayoutManager.scrollBy(LinearLayoutManager.java:1332)
                                                                            at android.support.v7.widget.LinearLayoutManager.scrollVerticallyBy(LinearLayoutManager.java:1076)
                                                                            at android.support.v7.widget.RecyclerView.scrollByInternal(RecyclerView.java:1790)
                                                                            at android.support.v7.widget.RecyclerView.onTouchEvent(RecyclerView.java:2996)
                                                                            at android.view.View.dispatchTouchEvent(View.java:10779)
                                                                            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2859)
                                                                            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2535)
                                                                            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2865)
                                                                            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2550)
                                                                            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2865)
                                                                            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2550)
                                                                            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2865)
                                                                            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2550)
                                                                            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2865)
                                                                            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2550)
                                                                            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2865)
                                                                            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2550)
                                                                            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2865)
                                                                            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2550)
                                                                            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2865)
                                                                            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2550)
                                                                            at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:582)
                                                                            at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1885)
                                                                            at android.app.Activity.dispatchTouchEvent(Activity.java:3240)
                                                                            at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:68)
                                                                            at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:544)
                                                                            at android.view.View.dispatchPointerEvent(View.java:11008)
                                                                            at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5162)
                                                                            at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5014)
                                                                            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4539)
                                                                            at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4592)
                                                                            at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4558)
                                                                            at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4691)
                                                                            at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4566)
                                                                            at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4748)
                                                                            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4539)
                                                                            at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4592)
                                                                            at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4558)
12-26 15:55:05.145 32400-32400/org.fossasia.openevent E/AndroidRuntime:     at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4566)
                                                                            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4539)
                                                                            at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7099)
                                                                            at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7031)
                                                                            at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6992)
                                                                            at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7209)
                                                                            at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
                                                                            at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(Native Method)
                                                                            at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:176)
                                                                            at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:7173)
                                                                            at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:7236)
                                                                            at android.view.Choreographer$CallbackRecord.run(Choreographer.java:959)
                                                                            at android.view.Choreographer.doCallbacks(Choreographer.java:734)
                                                                            at android.view.Choreographer.doFrame(Choreographer.java:664)
                                                                            at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:945)
                                                                            at android.os.Handler.handleCallback(Handler.java:751)
                                                                            at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                            at android.os.Looper.loop(Looper.java:154)
                                                                            at android.app.ActivityThread.main(ActivityThread.java:6776)
                                                                            at java.lang.reflect.Method.invoke(Native Method)
                                                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1496)
                                                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1386)

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by mayank8318**
Fixes #2063 

Changes: Added check for empty text query so that the app doesn't crash. 

Screenshots for the change:

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public boolean onCreateOptionsMenu(Menu menu) {
 
     @Override
     public boolean onQueryTextChange(String query) {
-        if (query != null) {
+        if (query != null && !query.isEmpty()) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java]: Use TextUtils.isEmpty

**Comment by mayank8318** [comment]
Changes : Use TextUtils.isEmpty()
@iamareebjamal Please review :)

File: android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java [hunk 2]
```diff
@@ -100,7 +100,7 @@ public boolean onCreateOptionsMenu(Menu menu) {
 
     @Override
     public boolean onQueryTextChange(String query) {
-        if (query != null) {
+        if (query != null && !TextUtils.isEmpty(query)) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java]: Hmm. I meant don't use null check as isEmpty already does that

**Comment by mayank8318** [comment]
Changes : Replace null and blank check with TextUtils.isEmpty()
@iamareebjamal Please review :)

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**